src/saveload/marker.rs

```diff
@@ -109,49 +110,92 @@ impl<'a> EntityBuilder<'a> {
 ///         marker
 ///     }
 ///
-///     fn get(&self, id: u64) -> Option<Entity> {
+///     fn retrieve_entity_internal(&self, id: u64) -> Option<Entity> {
 ///         self.mapping.get(&id).cloned()
 ///     }
+///
+///     fn maintain(&mut self, entities: &EntitiesRes, storage: &ReadStorage<NetMarker>) {
+///        self.mapping = (&*entities, storage)
+///            .join()
+///            .map(|(e, m)| (m.id(), e))
+///            .collect();
+///     }
 /// }
 ///
 /// fn main() {
 ///     let mut world = World::new();
 ///     world.register::<NetMarker>();
 ///
-///     let mut node = NetNode {
-///         range: 0..100,
-///         mapping: HashMap::new(),
-///     };
+///     world.add_resource(
+///         NetNode {
+///             range: 0..100,
+///             mapping: HashMap::new(),
+///         }
+///     );
 ///
-///     let entity = world.create_entity().build();
-///     let (marker, added) = node.mark(entity, &mut world.write::<NetMarker>());
-///     assert!(added);
+///     let entity = world.create_entity().marked::<NetMarker>().build();
+///     let storage = &mut world.write::<NetMarker>();
+///     let marker = storage.get(entity).unwrap().clone();
 ///     assert_eq!(
-///         node.get_marked(marker.id(), &world.entities(), &mut world.write::<NetMarker>()),
+///         world.write_resource::<NetNode>().retrieve_entity(marker, storage, &*world.entities()),
 ///         entity
 ///     );
 /// }
 /// ```
-pub trait Marker: Component + DeserializeOwned + Serialize + Copy {
+pub trait Marker
+    : Clone + Component + Debug + Eq + Hash + DeserializeOwned + Serialize {
     /// Id of the marker
-    type Identifier: Copy + Debug + Eq + Hash;
-
+    type Identifier;
     /// Allocator for this `Marker`
     type Allocator: MarkerAllocator<Self>;
 
-    /// Get this marker internal id
+    /// Get this marker internal id.
+    /// This value may never change.
```

Comment:
As the update method says that changes in return value doesn't cause UB, I think the wording should be relaxed.
Maybe something like: "The value of this method should be constant".